hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -170,37 +171,37 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
 
       // Need to check if any files exist in the given path, if they exist we
       // cannot create a directory with the given key.
+      // Verify the path against prefix table
       OMFileRequest.OMPathInfo omPathInfo =
-          OMFileRequest.verifyFilesInPath(omMetadataManager, volumeName,
```

Comment:
As we store the dir in prefix table now, we need to acquire PREFIX_LOCK instead of BUCKET_LOCK.
`
acquiredLock = omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
 volumeName, bucketName);
`

Reply:
Very good point. Will improve this logic.